src/plus/gk/serverConnection.ts

```diff
@@ -135,6 +155,9 @@ export class ServerConnection implements Disposable {
 	}
 
 	private async gkFetch(url: RequestInfo, init?: RequestInit, options?: GKFetchOptions): Promise<Response> {
+		if (this.requestsAreBlocked) {
+			throw new RequestsAreBlockedTemporarilyError();
```

Comment:
We will want to keep an eye on any pathways to `gkFetch` where the caller is not wrapping the call (i.e. `fetchApi` or `fetchGkDevApi`) in a try/catch block and handling a thrown error. For example, `cloudIntegrationService` has two such calls not wrapped in a try/catch block, and I'm not certain what will happen if these fail to catch errors (and it throws up the caller chain).

Not suggesting that there is an issue here - there may be existing cases where these calls throw. Just giving a heads-up that we're introducing one new possibility where a call returns with `rsp.ok` as `false` five times, triggers our temporary block, and then on that same call we start throwing errors (as we should). Hopefully these errors throwing up the chain does not cause issues.

Reply:
@axosoft-ramint 
Maybe wrap there? There not-ok-handling logs and sends telemetry events, then returns `undefined`. We can catch, log, and return `undefined`. (I suppose we don't need to telemetry here)

Reply:
Yes, that sounds right. Can be done outside of this PR - perhaps we can write an issue for it.